File: 2679. Sum in a Matrix/Solution.py
import heapq

class Solution:
    def matrixSum(self, nums: List[List[int]]) -> int:
        r, c, res = len(nums), len(nums[0]), 0
        pqs = [[] for i in range(r)]
        for i in range(r):
            for j in range(c):
                heappush(pqs[i], -nums[i][j])
        for i in range(c):
            high = 0
            for j in range(r):
                high = max(high, -heappop(pqs[j]))
            res += high
        return res

File: 2679. Sum in a Matrix/Solution.java
class Solution {
    public int matrixSum(int[][] nums) {
        int r = nums.length, c = nums[0].length, res = 0;
        PriorityQueue<Integer>[] pqs = new PriorityQueue[r];
        for (int i = 0; i < r; i++) {
            pqs[i] = new PriorityQueue<>((a, b) -> b - a);
            for (int j = 0; j < c; j++) pqs[i].offer(nums[i][j]);
        }
        for (int i = 0; i < c; i++) {
            int high = 0;
            for (int j = 0; j < r; j++) high = Math.max(high, pqs[j].poll());
            res += high;
        }
        return res;
    }
}